schematic/store/synapse.py

```diff
@@ -100,14 +101,35 @@ def __init__(
 
         self._query_fileview()
 
+    def _purge_synapse_cache(self):
+        '''
+        Purge synapse cache if it exceeds 19.5 GB
+        '''
+        # try clearing the cache
+        # scan a directory and check size of files
+        root_dir = self.syn.cache.cache_root_dir
+        if os.path.exists(root_dir):
+            nbytes = get_dir_size(root_dir)
+            # if 19.5 GB has already been taken, purge cache prior to today
+            if nbytes >= 20937965568:
+                c = self.syn.core.cache.Cache()
```

Comment:
@thomasyu888 when I tested the purge functionality today, I noticed that this line would trigger an attribute error. I think `cache = self.syn.cache` and `cache.purge()` would work